ts/core/MmlTree/MmlNodes/mstyle.ts

```diff
@@ -75,10 +75,16 @@ export class MmlMstyle extends AbstractMmlLayoutNode {
       } else {
         level = parseInt(scriptlevel);
       }
+      prime = false;  // style change resets tex prime style
     }
     let displaystyle = this.attributes.getExplicit('displaystyle') as boolean;
     if (displaystyle != null) {
       display = (displaystyle === true);
+      prime = false;  // style change resets tex prime style
+    }
+    const cramped = this.attributes.getExplicit('data-cramped') as boolean;  // manual control of tex prime style
+    if (cramped != null) {
+      prime = cramped;
```

Comment:
You want to set `prime` only if `cramped` is not `undefined` or `null`? Can't you just set it to false in all of these cases? (i.e., `prime = !!cramped`)?

Reply:
Not quite.  Here we are only adjusting the prime style if cramped is explicitly given, otherwise we retain the current prime style.  We don't reset it to false if the `data-cramped` attribute is not present.  So the check for not null is important.